frontend/src/scenes/persons/activityDescriptions.tsx

```diff
@@ -0,0 +1,112 @@
+import { ActivityChange, ActivityLogItem } from 'lib/components/ActivityLog/humanizeActivity'
+import React from 'react'
+import { PersonType } from '~/types'
+import { PropertyKeyInfo } from 'lib/components/PropertyKeyInfo'
+import { PersonHeader } from 'scenes/persons/PersonHeader'
+import { SentenceList } from 'scenes/feature-flags/activityDescriptions'
+
+type personFields = keyof PersonType
+
+const isRecord = (candidate?: string | Record<string, any> | boolean): candidate is Record<string, any> =>
+    typeof candidate === 'object'
+
+const personActionsMapping: {
+    [field in personFields]: (item: ActivityLogItem, change?: ActivityChange) => string | JSX.Element | null
+} = {
+    properties: function onChangedProperty(_, change) {
+        const before = change?.before
+        const after = change?.after
+
+        if (!after || !isRecord(after)) {
+            // must always have an after which is a record
+            return null
+        }
+
+        // UI only lets you add properties one at a time,
+        // so this can return the first changed property
+        // there should always and only be one... but it depends on how often the API is used to patch Persons
+        const changedProperties = Object.keys(after).filter((key) => isRecord(before) && !(key in before))
+        if (changedProperties[0]) {
+            return (
+                <>
+                    added property{' '}
+                    <strong>
+                        <PropertyKeyInfo value={changedProperties[0]} />
+                    </strong>{' '}
+                    with value:{' '}
+                    <strong>
+                        {after[changedProperties[0]] === null ? 'null' : String(after[changedProperties[0]])}
+                    </strong>
+                </>
+            )
+        }
+
+        return null
+    },
+    // fields that shouldn't show in the log if they change
+    id: () => null,
+    uuid: () => null,
+    distinct_ids: () => null,
+    ['name']: () => null,
+    created_at: () => null,
+    is_identified: () => null,
```

Comment:
Can't we just not log the changes then? Or have another `display = False` attribute on the activity log? Removing rows at will in the frontend feels like a bad idea for many reasons. Mostly pagination.

Reply:
🤔